### PR TITLE
In preparation for the removal of advertised services, @CordaService …

### PR DESCRIPTION
**Pull request by shamsasari**
…no longer expects a static "type" field for the ServiceType.

Instead @CordaServices will use the main identity of the node.

### CODE REVIEW

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt [hunk 1]
```diff
@@ -45,7 +45,7 @@ import java.util.*
  * A singleton utility that only provides a mock identity, key and storage service. However, this is sufficient for
  * building chains of transactions and verifying them. It isn't sufficient for testing flows however.
  */
-open class MockServices(vararg val keys: KeyPair) : ServiceHub {
+open class MockServices(vararg val keys: KeyPair) : PluginServiceHub {
```
Comment (kasiastreich) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt]: Isn't  `PluginServiceHub` holding only deprecated method `registerFlowInitiator`?

File: finance/src/main/kotlin/net/corda/finance/contracts/FinanceTypes.kt [hunk 2]
```diff
@@ -416,7 +415,7 @@ interface FixableDealState : DealState {
     /**
      * What oracle service to use for the fixing
      */
-    val oracleType: ServiceType
+    val oracle: Party
```
Comment (kasiastreich) [inline finance/src/main/kotlin/net/corda/finance/contracts/FinanceTypes.kt]: I started very similar changes in services removal. Thanks for clean up!

File: node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt [hunk 3]
```diff
@@ -135,9 +135,6 @@ abstract class AbstractNode(open val configuration: NodeConfiguration,
     lateinit var database: CordaPersistence
     protected var dbCloser: (() -> Any?)? = null
 
-    var isPreviousCheckpointsPresent = false
```
Comment (kasiastreich) [inline node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt]: Is this service removal related?

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt (hunk 1 above)
Comment (shamsasari) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt]: Yes, this change was to make it easier to work with a test. PluginServiceHub itself probably can be removed

File: node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt (hunk 3 above)
Comment (shamsasari) [reply node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt]: No, just code cleanup
